skins/oasis/modules/BodyController.class.php

```diff
@@ -77,286 +111,309 @@ public static function isGridLayoutEnabled() {
+	 *
+	 * @return bool
 	 */
 	public static function showUserPagesHeader() {
-		wfProfileIn(__METHOD__);
-
-		global $wgTitle;
+		$app = F::app();
+		$wgTitle = $app->wg->Title;
 
 		// perform namespace and special page check
-		$isUserPage = in_array($wgTitle->getNamespace(), self::getUserPagesNamespaces());
-
-		$ret = ($isUserPage && !$wgTitle->isSubpage() )
-				|| $wgTitle->isSpecial( 'Following' )
-				|| $wgTitle->isSpecial( 'Contributions' )
-				|| $wgTitle->isSpecial( 'UserActivity' )
-				|| (defined('NS_BLOG_LISTING') && $wgTitle->getNamespace() == NS_BLOG_LISTING)
-				|| (defined('NS_BLOG_ARTICLE') && $wgTitle->getNamespace() == NS_BLOG_ARTICLE);
+		$isUserPage = in_array(
+			$wgTitle->getNamespace(),
+			self::getUserPagesNamespaces()
+		);
+
+		$ret = ( $isUserPage && !$wgTitle->isSubpage() ) ||
+				$wgTitle->isSpecial( 'Following' ) ||
+				$wgTitle->isSpecial( 'Contributions' ) ||
+				$wgTitle->isSpecial( 'UserActivity' ) ||
+				(
+					defined( 'NS_BLOG_LISTING' ) &&
+					$wgTitle->getNamespace() == NS_BLOG_LISTING
+				) ||
+				(
+					defined( 'NS_BLOG_ARTICLE' ) &&
+					$wgTitle->getNamespace() == NS_BLOG_ARTICLE
+				);
 
-		wfProfileOut(__METHOD__);
 		return $ret;
 	}
 
 	/**
 	 * Return list of namespaces on which user pages header should be shown
+	 *
+	 * @return array $namespaces
 	 */
 	public static function getUserPagesNamespaces() {
-		global $wgEnableWallExt;
+		$app = F::app();
+		$namespaces = [ NS_USER ];
 
-		$namespaces = array(NS_USER);
-		if( empty($wgEnableWallExt) ) {
+		if ( empty( $app->wg->EnableWallExt ) ) {
 			$namespaces[] = NS_USER_TALK;
 		}
-		if (defined('NS_BLOG_ARTICLE')) {
+
+		if ( defined( 'NS_BLOG_ARTICLE' ) ) {
 			$namespaces[] = NS_BLOG_ARTICLE;
 		}
-		if (defined('NS_BLOG_LISTING')) {
-			// FIXME: THIS IS NOT REALLY PART OF THE USER PAGES NAMESPACES
-			//$namespaces[] = NS_BLOG_LISTING;
+
+		/*
+		// FIXME: THIS IS NOT REALLY PART OF THE USER PAGES NAMESPACES
+		if ( defined( 'NS_BLOG_LISTING' ) ) {		
+			$namespaces[] = NS_BLOG_LISTING;
 		}
-		if (defined('NS_USER_WALL')) {
+		*/
+
+		if ( defined( 'NS_USER_WALL' ) ) {
 			$namespaces[] = NS_USER_WALL;
 		}
+
 		return $namespaces;
 	}
 
+	/**
+	 * Get a list of rail modules
+	 *
+	 * @return array $railModuleList
+	 */
 	public function getRailModuleList() {
-		wfProfileIn(__METHOD__);
-		global $wgTitle, $wgUser, $wgEnableAchievementsExt, $wgContentNamespaces,
-			$wgExtraNamespaces, $wgExtraNamespacesLocal,
-			$wgEnableWikiAnswers, $wgEnableHuluVideoPanel,
-			$wgEnableWallEngine, $wgRequest,
-			$wgEnableForumExt;
-
-		$namespace = $wgTitle->getNamespace();
+		$wg = $this->wg;
+
+		$namespace = $wg->Title->getNamespace();
 		$subjectNamespace = MWNamespace::getSubject($namespace);
 
-		$railModuleList = array();
+		$railModuleList = [];
+
+		// temp plocation for testing
+		// @todo figure out where this should go
+		$railModuleList[1460] = [ 'Contribute', 'index', null ];
```

Comment:
Wouldn't 1490+ be above everything in the right rail? I thought it needed to be somewhere between PagesOnWiki (1450) and Ad (1490)?